feat(bext): add ability to remove saved sg url from suggestion list

erzhtor opened the pull request:
## Description

This PR adds ability to remove saved sourcegraph url from the suggestion list.

## Test plan
- `pnpm run storybook:browser`
- Open http://localhost:9001/?path=/story/browser-options-optionspage--all-options-pages
<!--
  As part of SOC2/GN-104 and SOC2/GN-105 requirements, all pull requests are REQUIRED to
  provide a "test plan". A test plan is a loose explanation of what you have done or
  implemented to test this, as outlined in our Testing principles and guidelines:
  https://docs.sourcegraph.com/dev/background-information/testing_principles
  Write your test plan here after the "## Test plan" header.
-->

## Screenshots

https://github.com/sourcegraph/sourcegraph/assets/6717049/028317bb-570a-42a6-8b7d-ecdc00acdb8e

## Before merging

- [ ] Test on different code hosts (if applicable)
    - [ ] GitHub
    - [ ] Gitlab
    - [ ] GitHub Enterprise
    - [ ] Refined GitHub
    - [ ] Phabricator
    - [ ] Phabricator integration
    - [ ] Bitbucket
    - [ ] Bitbucket integration

- [ ] Test on different browsers (if applicable)
    - [ ] Chrome
    - [ ] Firefox
    - [ ] Safari
- [ ] Add change log message to [client/browser/CHANGELOG.md](./client/browser/CHANGELOG.md), under "Unreleased" section. (if applicable)

vovakulikov reviewed (commented):
The code looks ok. A few suggestions about UX/UI 
- Delete buttons alignment looks a bit off. I understand this is happening because of our design system spacing problems. I would suggest adjusting padding and making them similar to what we did for the delete button in the MultiCombobox pills block 
<img width="396" alt="Screenshot 2023-05-29 at 12 33 27" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/906a6b98-980e-41ff-8530-3507a90ea9f6">

- Tooltip "Remove suggestions" feels unnecessary here because it interrupts cursors if you try to hover the delete button below (this is actually a [correct behaviour for the tooltip](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html), it should be hoverable), but in this case, I think we can drop tooltip because delete cursor flow is more important (and we have confirmation dialogue after, so it should be clear of what you're doing)


https://github.com/sourcegraph/sourcegraph/assets/18492575/b282ce21-0384-4588-b4ca-f78472ee3a9b

erzhtor commented:
@vovakulikov, thanks for the review. 

> * Tooltip "Remove suggestions" feels unnecessary here because it interrupts cursors if you try to hover the delete button below (this is actually a [correct behaviour for the tooltip](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html), it should be hoverable), but in this case, I think we can drop tooltip because delete cursor flow is more important (and we have confirmation dialogue after, so it should be clear of what you're doing)

Agree, removed the tooltip.

> * Delete buttons alignment looks a bit off. I understand this is happening because of our design system spacing problems. I would suggest adjusting padding and making them similar to what we did for the delete button in the MultiCombobox pills block
> <img alt="Screenshot 2023-05-29 at 12 33 27" width="396" src="https://user-images.githubusercontent.com/18492575/241764643-906a6b98-980e-41ff-8530-3507a90ea9f6.png">

Good point, thanks for pointing that out. Updated.

vovakulikov reviewed (approved):
Looks cool, thanks you!
